tests/test_main_addheader.py

```diff
@@ -939,3 +939,59 @@ def test_addheader_line_endings(
         .replace("spdx", "SPDX")
         .replace("\n", line_ending)
     )
+
+
+def test_addheader_recursive(fake_repository, stringio, mock_date_today):
+    """Add a header to a directory recursively."""
+    (fake_repository / "src/one/two").mkdir(parents=True)
+    (fake_repository / "src/one/two/foo.py").write_text(
+        cleandoc(
+            """
+            # spdx-License-Identifier: GPL-3.0-or-later
+            """
+        ).replace("spdx", "SPDX")
+    )
+    (fake_repository / "bar").mkdir(parents=True)
+    (fake_repository / "bar/bar.py").touch()
```

Comment:
Just a note on what I detected during testing: reuse addheader (at least recursively) does not have an effect on empty files. It only picked them up when some content was in it. So this test may not catch all that you want (also making sure that nothing is written into `bar/bar.py`)